### PR TITLE
Adding SubArray tests to generic.jl, includes fix for linreg bug found

### PR DESCRIPTION
**Pull request by jacobflath**
Adding tests to the generic.jl test file for subarrays. Tests discovered
bug in the method Linreg in generic.jl, commit includes fix for that bug.

Tests added in due to need references in issue JuliaLang/LinearAlgebra.jl#299

### CODE REVIEW

File: test/linalg/generic.jl [hunk 1]
```diff
@@ -89,42 +95,76 @@ y = ['a','b','c','d','e']
...
-@test scale!(similar(a), [1.; 2.], a) == a.*[1; 2]
-@test scale!(similar(a), a, [1; 2; 3]) == a.*[1 2 3]
-@test scale!(similar(a), [1; 2], a) == a.*[1; 2]
-@test_throws DimensionMismatch scale!(similar(a), a, ones(2))
-@test_throws DimensionMismatch scale!(similar(a), ones(3), a)
-@test_throws DimensionMismatch scale!(Array(Float64, 3, 2), a, ones(3))
+
+# array and subarray tests
+let aa = reshape([1.:6;], (2,3))
+    for atype in ("Array", "SubArray")
```
Comment (andreasnoack) [inline test/linalg/generic.jl]: Minor detail, but maybe just use the types here instead of strings, i.e. `Array` and `SubArray`

**Comment by andreasnoack** [comment]
@jacobflath Welcome to Julia and thanks for adding these tests.

File: test/linalg/generic.jl (hunk 1 above)
Comment (jacobflath) [inline test/linalg/generic.jl]: I can definitely do that, I was just attempting to follow the precedent set in the files where the SubArray tests are already added

Comment (andreasnoack) [inline test/linalg/generic.jl]: It doesn't really matter so let's just leave it as it is.

Comment (jacobflath) [inline test/linalg/generic.jl]: Okay, sounds good

**Pull request merged**